teambit/bit
# Upgrade webpack

## Proposed Changes

- upgrade webpack, webpack-dev-server, loaders, 
- fix breaking changes from WebpackDevSever
- fix css hot reloading (by replacing MiniCssExtractPlugin with style-loader in dev mode)
- fix jsx hot reloading
- might fix fonts loading (#4816, #4805, #4693)
- fixes, types

Tested:
- [x] hot reloading jsx
- [x] hot reloading css 
- [x] fonts, svg,
- [x] errors overlay (but not source file errors)
- [x] tag & export

Follow up (in a separate PR):
- Extract css loaders to their own component, and simplify
- Remove our custom overlays for React Fast Refresh
- Apply the same updates to the main UI
- Check how hot reloading continued working even after removing react-refresh/babel (😵)
